sys-boot/udk/udk-2018.ebuild

```diff
@@ -0,0 +1,217 @@
+	elif tc-is-clang; then
+		TOOLCHAIN_TAG="CLANG38"
+	else
+		TOOLCHAIN_TAG="ELFGCC"
+	fi
+}
+
+src_unpack() {
+	default
+
+	local doc_name
+	local f
+	if use doc; then
+		pushd "${DOCDIR}" >/dev/null || die
+		rm -f *.chm || die
+		for f in *.zip; do
+			unpack "${DOCDIR}/${f}"
+			mv html "${f%.zip}" || die
+		done
+		rm -f *.zip || die
+		popd >/dev/null || die
+	fi
+}
+
+src_configure() {
+	# Compile of Base Tools is required for further setting up the environment
+	# Base tools does not like parallel make
+	sed -e "s:^\(BUILD_CFLAGS\s*=\).*$:\1 ${CFLAGS} -MD -fshort-wchar -fno-strict-aliasing -nostdlib -c -fPIC:" \
+		-i "BaseTools/Source/C/Makefiles/header.makefile" \
+		|| die "Failed to update makefile header"
+	local make_flags=(
+		BUILD_CC="$(tc-getBUILD_CC)"
+		BUILD_CXX="$(tc-getBUILD_CXX)"
+		BUILD_AS="$(tc-getBUILD_AS)"
+		BUILD_AR="$(tc-getBUILD_AR)"
+		BUILD_LD="$(tc-getBUILD_LD)"
+	)
+	emake "${make_flags[@]}" -j1 -C BaseTools
+	. edksetup.sh
+
+	# Update UDK parameter files
+	sed -e "s:^\(ACTIVE_PLATFORM\s*=\).*$:\1 MdeModulePkg/MdeModulePkg.dsc:" \
+		-e "s:^\(TARGET\s*=\).*$:\1 RELEASE:" \
+		-e "s:^\(TARGET_ARCH\s*=\).*$:\1 ${ARCH}:" \
+		-e "s:^\(TOOL_CHAIN_TAG\s*=\).*$:\1 ${TOOLCHAIN_TAG}:" \
+		-e "s:^\(MAX_CONCURRENT_THREAD_NUMBER\s*=\).*$:\1 $(makeopts_jobs):" \
+		-i "Conf/target.txt" || die "Failed to configure target file"
+	sed -e "s:\(_\(CC\|ASM\|PP\|VFRPP\|ASLCC\|ASLPP\|DLINK\)_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASLDLINK_PATH\s*=\).*$:\1 $(tc-getLD):" \
+		-e "s:\(_OBJCOPY_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+		-e "s:\(_RC_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+		-e "s:\(_SLINK_PATH\s*=\).*$:\1 $(tc-getAR):" \
+		-i "Conf/tools_def.txt" \
+		|| die "Failed to prepare tools definition file"
+}
+
+src_compile() {
+	local build_target
+	if use examples; then
+		build_target=all
+	else
+		build_target=libraries
+	fi
+
+	build ${build_target} || die "Failed to compile environment"
+
+# TODO * /var/tmp/portage/sys-apps/sandbox-2.10-r1/work/sandbox-2.10/libsandbox/trace.c:_do_ptrace():74: failure (Operation not permitted):
+# TODO * ISE:_do_ptrace: ptrace(PTRACE_TRACEME, ..., 0x0000000000000000, 0x0000000000000000): Operation not permitted
+}
+
+src_install() {
+	local f
+	local build_dir="${S}/Build/MdeModule/RELEASE_${TOOLCHAIN_TAG}/${ARCH}"
+
+	for f in "${build_dir}"/*/Library/*/*/OUTPUT/*.lib; do
+		local fn="lib${f##*/}"
+		newlib.a "${f}" "${fn%.lib}.a"
+	done
+
+	insinto "/usr/share/${P}"
+	doins "${S}/BaseTools/Scripts/GccBase.lds"
+
+	insinto "/usr/include/${PN}"
+	doins "${S}/MdePkg/Include/"*.h
+	doins -r "${S}/MdePkg/Include/"{${ARCH}/.,Guid,IndustryStandard,Library,Pi,Ppi,Protocol,Uefi}
+	local hfile
+	while read -d '' -r hfile; do
+		doins -r "${hfile}/."
+	done < <(find "${S}" -name 'BaseTools' -prune -o -name 'MdePkg' -prune -o \
+		-name 'CryptoPkg' -prune -o -type d -name Include -print0)
+
+	dobin "${S}/BaseTools/Source/C/bin/GenFw"
+
+	HTML_DOCS="${DOCDIR}"/* einstalldocs
```

Comment:
`local HTML_DOCS=( "${DOCDIR}"/. )` maybe?

Reply:
Using it in front of einstalldocs is more restrictive than local, isn't it? So it does not “pollute” the rest of the function.
What is the difference between your suggestion and directly doing `HTML_DOCS="${DOCDIR}"`?

Reply:
Non-array syntax is deprecated and doesn't work well with spaces in paths. Yeah, I know those are going to fail in another thousand places but better to avoid making it even worse.